libs/src/sdk/sdk.ts

```diff
@@ -153,18 +153,22 @@ const initializeApis = ({
 }) => {
   const initializationPromise = discoveryProvider.init()
 
-  const fetchApi = async (url: string) => {
+  const fetchApi = async (url: string, context?: RequestInit) => {
     // Ensure discovery node is initialized
     await initializationPromise
 
     // Append the appName to the query params
     const urlWithAppName =
       url + (url.includes('?') ? '&' : '?') + querystring({ app_name: appName })
 
+    const requestParams: Record<string, unknown> = {
+      endpoint: urlWithAppName
+    }
+    if (context !== undefined) {
+      requestParams['headers'] = context['headers']
```

Comment:
why not obj assign request params to context and _then_ add the endpoint? Rather than only including headers?